js/entityManager.ts

```diff
@@ -15,30 +15,33 @@ export class EntityManager {
   handleLoadedIndex(entity: Entity, event: Event): void {
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore
-    const { response } = (event as ProgressEvent<
+    const { status, response } = (event as ProgressEvent<
       XMLHttpRequestEventTarget
     >).target;
 
-    this.wasmQueue.runAfterWasmLoaded(() => {
-      const indexInfo = wasmRegisterIndex(
-        entity.name,
-        new Uint8Array(response)
-      );
+    if (status < 200 || status > 299) {
+      entity.setDownloadError();
+      throw new Error(`Got a ${status} error from ${entity.url}!`)
+    }
 
-      // Force end progress after index is registered
-      entity.setDownloadProgress(1);
+    this.wasmQueue.runAfterWasmLoaded(() => {
+      if (!entity.error) {
+        const indexInfo = wasmRegisterIndex(
+          entity.name,
+          new Uint8Array(response)
+        );
 
-      entity.performSearch(entity.domManager.getQuery());
+        entity.setDownloadProgress(1);
+        entity.performSearch(entity.domManager.getQuery());
 
-      if (entity.config.printIndexInfo) {
-        this.wasmQueue.runAfterWasmLoaded(() => {
```

Comment:
This seemed... completely unnecessary, so I'm removing it. This might come back to bite me, but hopefully it doesn't!